models/istio_rule_test.go

```diff
@@ -0,0 +1,148 @@
+package models
+
+import (
+	"testing"
+
+	"github.com/kiali/swscore/kubernetes"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIstioRulesParsing(t *testing.T) {
+	assert := assert.New(t)
+
+	istioRules := CastIstioRulesCollection(fakeIstioRules())
+	assert.Equal(2, len(istioRules))
+
+	assert.Equal("checkfromcustomer", istioRules[0].Name)
+	assert.Equal("destination.labels[\"app\"] == \"preference\"", istioRules[0].Match)
+	actions, ok := (istioRules[0].Actions).([]map[string]interface{})
+	assert.True(ok)
```

Comment:
Perhaps it can be used the following method to assert a values type:
https://godoc.org/github.com/stretchr/testify/assert#Implements

```go
assert.Implements([]map[string]interface{}, istioRules[0].Actions)
```

Reply:
I haven't tested it with interfaces, but worth to try it.

Reply:
I looked good, but after tested it, I guess assert.Implements() is used better to validate that an object implements a specific interface, but on this case it doesn't validate that the type cast is performed correctly.

```
// Implements asserts that an object is implemented by the specified interface.
//
//    a.Implements((*MyInterface)(nil), new(MyObject))
```
So, I did a try locally but it didn't work as expected, so I would maintain this as it is as I guess is more clear in what it is being tested.